Batch of Code Review docs refactoring

dfarrell07 opened the pull request.

nyechiel reviewed (approved):
A separate commit for each change. Someone leads by example ;)

skitt commented on src/content/contributing/code-review/_index.en.md:
```diff
@@ -6,44 +6,39 @@ weight: 30
 Kubernetes recommends [avoiding merge commits][merge commits].
 
-Use `git fetch` and `git rebase` to avoid them.
+Our current GitHub configuration should automatically remove merge commits.
```
```suggestion
Our current GitHub configuration should automatically remove merge commits.
```

With our current GitHub setup, pull requests are liable to include merge commits temporarily: whenever a PR is updated through the UI, GitHub merges the target branch into the PR. However, since we merge PRs by either squashing or rebasing them, those merge commits disappear from the series of commits which ultimately ends up in the target branch.

skitt reviewed (changes requested):
A suggestion regarding merge commits, see the specific comment.

One aspect that’s missing here is “who?”; this doesn’t have to be added as part of this PR, and I can take care of it — for each point here, I think it would be useful to say who is responsible (the PR author, the reviewer(s), and/or the PR merger).

dfarrell07 commented:
@skitt Agreed it would be useful to add a "who" wherever possible.

It would also be nice to clarify how to accept suggestions on PRs without breaking the DCO check. I actually don't know. :sweat_smile:

dfarrell07 commented:
```
src/content/contributing/code-review/_index.en.md: 24: MD013/line-length Line length [Expected: 140; Actual: 179]
src/content/contributing/code-review/_index.en.md: 25: MD013/line-length Line length [Expected: 140; Actual: 172]
```

Fixing manually

tpantelis commented on src/content/contributing/code-review/_index.en.md:
```diff
@@ -6,44 +6,41 @@ weight: 30
 Kubernetes recommends [avoiding merge commits][merge commits].
 
-Use `git fetch` and `git rebase` to avoid them.
+With our current GitHub setup, pull requests are liable to include merge commits temporarily: whenever a PR is updated through the UI,
```
```suggestion
With our current GitHub setup, pull requests are liable to include merge commits temporarily. Whenever a PR is updated through the UI,
```

tpantelis commented on src/content/contributing/code-review/_index.en.md:
```diff
@@ -6,44 +6,41 @@ weight: 30
-milestones or units of work. Use commits to add clarity to the development and
-review process.
+All commits left on your branch after a review should represent meaningful milestones or units of work. Use commits to add clarity to the
+development and review process.
```
To further expand:

```suggestion
development and review process. Keep in mind that smaller commits are easier to review. In addition. when addressing review comments, as a general rule, push a new commit instead of amending to the prior commit as the former makes it easy for reviewers to determine what changed.
```

skitt replied:
Following on Tom’s suggestion: 

```
But remember to squash them into the appropriate commit before merging. To make that simpler, commit with `git commit --fixup`, with the appropriate commit hash: this will keep your review-prompted changes as separate commits, but automatically select them for squashing with `git rebase --autosquash -i`.
```

dfarrell07 commented:
I'm getting confused as to what you guys want @skitt and @tpantelis, and I think the scope of this section is creeping.

dfarrell07 commented:
> when addressing review comments, as a general rule, push a new commit instead of amending to the prior commit as the former makes it easy for reviewers to determine what changed.

> But remember to squash them into the appropriate commit before merging. To make that simpler, commit with `git commit --fixup`, with the appropriate commit hash: this will keep your review-prompted changes as separate commits, but automatically select them for squashing with `git rebase --autosquash -i`

I think both of these ideas are good to capture, but I don't think they should be part of the section about commits being discrete steps. I think they should be a new section about handling commits from PR feedback.

tpantelis commented:
> 
> I think both of these ideas are good to capture, but I don't think they should be part of the section about commits being discrete steps. I think they should be a new section about handling commits from PR feedback.

I didn't know that advanced stuff - I've captured it in my own notes for future reference. 😄

tpantelis commented on src/content/contributing/code-review/_index.en.md:
```diff
@@ -6,44 +6,50 @@ weight: 30
+When addressing review comments, as a general rule, push a new commit instead of amending to the prior commit as the former makes it easy
+for reviewers to determine what changed.
+
+But remember to squash them into the appropriate commit before merging. To make that simpler, commit with `git commit --fixup`, with the
+appropriate commit hash: this will keep your review-prompted changes as separate commits, but automatically select them for squashing with
+`git rebase --autosquash -i`.
```
Not trying to nit but expanded the content a bit - hope it's all accurate.

```suggestion
To avoid cluttering the git log, squash the review commits into the appropriate commit before merging. The committer can do this in GitHub via the "Squash and merge" option.  However you may want to preserve other commits, in which case, squashing will need to be done manually via the Git CLI. To make that simpler, you can commit the review-prompted changes with `git commit --fixup` with the
appropriate commit hash. This will keep them as separate commits and, when you later rebase, specify the `--autosquash` option (that is `git rebase --autosquash -i`) to automatically select them for squashing.
```

nyechiel reviewed (approved):
LGTM
